railties/lib/rails/templates/rails/mailers/email.html.erb

```diff
@@ -61,14 +61,14 @@
 <body>
 <header>
   <dl>
-    <% if @email.respond_to?(:smtp_envelope_from) && Array(@email.from) != Array(@email.smtp_envelope_from) %>
+    <% if @email.from != [@email.smtp_envelope_from] %>
       <dt>SMTP-From:</dt>
       <dd id="smtp_from"><%= @email.smtp_envelope_from %></dd>
     <% end %>
 
-    <% if @email.respond_to?(:smtp_envelope_to) && @email.to != @email.smtp_envelope_to %>
+    <% if Set.new(@email.to) + Set.new(@email.cc) + Set.new(@email.bcc) != Set.new(@email.smtp_envelope_to) %>
```

Comment:
We can clean this up a bit by using `Set[]`, which is somewhat comparable to `Array()`.

```suggestion
    <% if Set[*@email.to, *@email.cc, *@email.bcc] != Set[*@email.smtp_envelope_to] %>
```